module/Finna/src/Finna/RecordDriver/SolrLrmi.php

```diff
@@ -33,9 +37,465 @@
+     *
+     * @return mixed array with keys:
+     *   'copyright'   Copyright (e.g. 'CC BY 4.0')
+     *   'link'        Link to copyright info, see IndexRecord::getRightsLink
+     *   or false if no access restriction type is defined.
+     */
+    public function getAccessRestrictionsType()
+    {
+        $xml = $this->getXmlRecord();
+        $rights = [];
+        $locale = $this->getLocale();
+        if (!empty($xml->rights)) {
+            $copyrights = (string)$xml->rights;
+            $rights['copyrights']
+                = $this->usageRightsMap[$copyrights] ?? $copyrights;
+            $rights['link'] = $this->getRightsLink($rights['copyrights'], $locale);
+            return $rights;
+        }
+        return false;
+    }
+
+    /**
+     * Get an array of summary strings for the record
+     *
+     * @return array
+     */
+    public function getSummary()
+    {
+        $xml = $this->getXmlRecord();
+        $locale = $this->getLocale();
+        foreach ($xml->description as $d) {
+            if (!empty($d['format'])) {
+                continue;
+            }
+            if ($locale === (string)$d['lang']) {
+                return [(string)$d];
+            }
+        }
+        return [];
+    }
+
+    /**
+     * Get educational audiences
+     *
+     * @return array
+     */
+    public function getEducationalAudiences()
+    {
+        return $this->fields['educational_audience_str_mv'] ?? [];
+    }
+
+    /**
+     * Get all authors apart from presenters
+     *
+     * @return array
+     */
+    public function getNonPresenterAuthors()
+    {
+        $xml = $this->getXmlRecord();
+        $result = [];
+        foreach ($xml->author->person ?? [] as $author) {
+            $result[] = [
+                'name' => trim((string)$author->name),
+                'affiliation' => trim((string)$author->affiliation)
+            ];
+        }
+        foreach ($xml->author->organization ?? [] as $org) {
+            $result[] = [
+                'name' => trim((string)$org->legalName)
+            ];
+        }
+        return $result;
+    }
+
+    /**
+     * Return educational levels
+     *
+     * @return array
+     */
+    public function getEducationalLevels()
+    {
+        return $this->fields['educational_level_str_mv'] ?? [];
+    }
+
+    /**
+     * Return root educational levels
+     *
+     * @return array
+     */
+    public function getRootEducationalLevels()
+    {
+        $rootLevels = [];
+        foreach ($this->fields['educational_level_str_mv'] ?? [] as $level) {
+            if (substr($level, 0, 1) === '0') {
+                $rootLevels[] = trim((string)$level);
+            }
+        }
+        return $rootLevels;
+    }
+
+    /**
+     * Return url to external LRMI record page based on the record ID
+     * or false if ID is not provided.
+     *
+     * @return string|boolean
+     */
+    public function getExternalLink()
+    {
+        $xml = $this->getXmlRecord();
```

Comment:
Tätä tarvitaan vasta jos linkkitemplaatti löytyy, eli pois tästä. Käytetään vain yhdessä kohdassa, eli voisi siinä kutsua suoraan `$this->getXmlRecord()->recordID`, mutta voi sen $xml-muuttuja selkeyden nimissä jättääkin.

Jos recordID on sama kuin getUniqueID():n palauttama pisteen jälkeinen osa, voisi myös käyttää sitä ja säästää XML-accessin.